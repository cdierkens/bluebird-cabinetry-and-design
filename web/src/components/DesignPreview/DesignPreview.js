import PropTypes from "prop-types";
import React from "react";
import { builder } from "../../lib/image-url";
import Container from "../container";
import * as styles from "./DesignPreview.module.css";

const BASE_IMAGE_SIZE = 300;

const DesignPreview = ({ title, description, images }) => {
  return (
    <section className="py-6">
      <Container className={styles.DesignContainer}>
        <h2 className={styles.DesignH2}>{!title || "Design Process"}</h2>
        <p className={styles.DesignText}>
          {!description ||
            "The process we use is transparent and we will deliver on our promises. Most people have had a bad experience with remodeling, or know someone who has. We are upfront about expectations, pricing, and treat clients the way we want to be treated."}
        </p>
      </Container>
      <Container>
        <div className="grid grid-cols-2 sm:grid-cols-2 md:grid-cols-6 gap-4 mx-auto content-center justify-center items-center">
          {images.map(({ image, colSpan, rowSpan }) => {
            let width =
              colSpan * BASE_IMAGE_SIZE + Math.max(colSpan - 1, 0) * 16;
            let height =
              rowSpan * BASE_IMAGE_SIZE + Math.max(rowSpan - 1, 0) * 16;

            return (
              <div
                key={image.file.asset.id}
                className={`md:col-span-${colSpan} row-span-${rowSpan} h-full`}
              >
                <img
                  src={builder
                    .image(image.file.asset.id)
                    .size(width, height)
                    .fit("clip")
                    .url()}
                  alt={image.description}
                  className="w-full h-full p-1 shadow-md"
                />
              </div>
            );
          })}
        </div>
      </Container>
    </section>
  );
};

DesignPreview.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  images: PropTypes.array,
};

export default DesignPreview;
