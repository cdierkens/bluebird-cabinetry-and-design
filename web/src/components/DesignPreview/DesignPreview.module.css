.DesignContainer {
  @apply py-8 grid grid-cols-8;
}

.DesignH2 {
  @apply text-center text-blue-dark col-span-8;
}

.DesignText {
  @apply text-center col-span-8 leading-loose;
}
