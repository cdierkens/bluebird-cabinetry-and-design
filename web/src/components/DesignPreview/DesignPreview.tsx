import { graphql } from "gatsby";
import React from "react";
import { DesignPreviewImageFragment } from "../../../graphql-types";
import { builder } from "../../lib/image-url";
import { invariant } from "../../lib/invariant";
import Container from "../Container";
import * as styles from "./DesignPreview.module.css";

const BASE_IMAGE_SIZE = 300;

export const query = graphql`
  fragment DesignPreview on SanityDesignPreview {
    title
    description
    images {
      ...DesignPreviewImage
    }
  }

  fragment DesignPreviewImage on SanityDesignPreviewImage {
    colSpan
    rowSpan
    image {
      description
      file {
        asset {
          id
        }
      }
    }
  }
`;
interface DesignPreviewProps {
  title: string;
  description: string;
  images: Array<DesignPreviewImageFragment>;
}

const DesignPreview: React.FC<React.PropsWithChildren<DesignPreviewProps>> = ({
  title,
  description,
  images,
}) => {
  return (
    <section className="py-6">
      <Container className={styles.DesignContainer}>
        <h2 className={styles.DesignH2}>{!title || "Design Process"}</h2>
        <p className={styles.DesignText}>
          {!description ||
            "The process we use is transparent and we will deliver on our promises. Most people have had a bad experience with remodeling, or know someone who has. We are upfront about expectations, pricing, and treat clients the way we want to be treated."}
        </p>
      </Container>
      <Container>
        <div className="grid grid-cols-2 sm:grid-cols-2 md:grid-cols-6 gap-4 mx-auto content-center justify-center items-center">
          {images.map(({ image, colSpan, rowSpan }) => {
            invariant(image?.file?.asset?.id);
            invariant(image?.file?.asset?.id);
            invariant(colSpan);
            invariant(rowSpan);

            let width =
              parseInt(colSpan, 10) * BASE_IMAGE_SIZE +
              Math.max(parseInt(colSpan, 10) - 1, 0) * 16;
            let height =
              parseInt(rowSpan, 10) * BASE_IMAGE_SIZE +
              Math.max(parseInt(rowSpan, 10) - 1, 0) * 16;

            return (
              <div
                key={image.file.asset.id}
                className={`md:col-span-${colSpan} row-span-${rowSpan} h-full`}
              >
                <img
                  src={
                    builder
                      .image(image.file.asset.id)
                      .size(width, height)
                      .fit("clip")
                      .url() ?? undefined
                  }
                  alt={image.description || undefined}
                  className="w-full h-full p-1 shadow-md"
                />
              </div>
            );
          })}
        </div>
      </Container>
    </section>
  );
};

export default DesignPreview;
