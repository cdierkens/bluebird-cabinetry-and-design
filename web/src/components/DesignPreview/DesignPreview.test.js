import { render } from "@testing-library/react";
import React from "react";
import DesignPreview from "./DesignPreview";

it("matches snapshot", () => {
  const { asFragment } = render(
    <DesignPreview
      images={[]}
      title="mock-title"
      description="mock-description"
    />
  );

  expect(asFragment()).toMatchSnapshot();
});
