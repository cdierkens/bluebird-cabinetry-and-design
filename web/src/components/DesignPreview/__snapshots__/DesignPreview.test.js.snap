// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <section
    class="py-6"
  >
    <div
      class="Container DesignContainer"
    >
      <h2
        class="DesignH2"
      >
        mock-title
      </h2>
      <p
        class="DesignText"
      >
        mock-description
      </p>
    </div>
    <div
      class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-6 gap-4 mx-auto w-2/3 xl:w-1/2 content-center justify-center items-center"
    />
  </section>
</DocumentFragment>
`;
